src/Prototype/Exe/Form/Signup.hs

```diff
@@ -0,0 +1,160 @@
+                      H.div
+                        ! A.class_ "o-form-group__controls"
+                        $ H.input
+                        ! A.class_ "c-input"
+                        ! A.id "email"
+                        ! A.name "email"
+                        ! A.type_ "email"
+                    H.div ! A.class_ "o-form-group" $ do
+                      H.label
+                        ! A.class_ "o-form-group__label"
+                        ! A.for "password"
+                        $ "Password"
+                      H.div
+                        ! A.class_ "o-form-group__controls"
+                        $ H.div
+                        ! A.class_ "c-input-with-icon"
+                        $ do
+                            H.input
+                              ! A.class_ "c-input"
+                              ! A.id "password"
+                              ! A.name "password"
+                              ! A.type_ "password"
+                            H.button
+                              ! A.class_ "c-input-with-icon__toggle"
+                              ! customAttribute "data-password-toggle" "1"
+                              $ do
+                                  H.div
+                                    ! A.class_ "o-svg-icon o-svg-icon-eye"
+                                    $ H.toMarkup svgIconEye
+                                  H.div
+                                    ! A.class_ "o-svg-icon o-svg-icon-eye-off"
+                                    $ H.toMarkup svgIconEyeOff
+                    H.div ! A.class_ "o-form-group" $ do
+                      H.div ! A.class_ "c-checkbox" $ do
+                        H.label $ do
+                          H.input
+                            ! A.id "i-understand"
+                            ! A.name "i-understand"
+                            ! A.type_ "checkbox"
+                          H.div $ do
+                            "I understand that this site is up for "
+                            "demonstration purpose only, and that data are "
+                            "regularly erased."
+                            H.a ! A.href "#" $ "Read more."
+                    H.div
+                      ! A.class_ "o-form-group"
+                      $ H.button
+                      ! A.class_ "c-button c-button--primary c-button--block"
+                      ! A.formaction signupSubmitURL
+                      ! A.formmethod "POST"
+                      $ H.span
+                      ! A.class_ "c-button__content"
+                      $ "Sign up"
+                    H.div ! A.class_ "o-form-group u-ta-center" $ ""
+        H.div ! A.class_ "c-content u-text-center u-spacer-top-l" $ do
+          H.a ! A.class_ "u-text-muted" ! A.href "#" $ "Log in"
+
+
+--------------------------------------------------------------------------------
+data ResultPage = Success Text
+                | Failure Text
+
```

Comment:
I personally think we should use our combinators for pages here; adding new ad-hoc types as we go is going to create a lot of contextual overhead and cruft.

Reply:
I think I agree. I was foreseeing that each form would have specific Success and Failure cases, but anyway this should be revisited as part of the work to get proper error handling of exceptions/`Nothing` cases.